tool/teleport/testing/test_server.go

```diff
@@ -14,7 +14,8 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-package mock
+// Package testing provides functions for creating test servers for testing.
+package testing
```

Comment:
This is a nasty name conflict with the standard "testing" package.

I suggest a different variant like `testenv` (which I do have in a few places) or `testonly` (even if you end up using an import alias in the end).

Reply:
+1 for testenv